independent-projects/arc/processor/src/main/java/io/quarkus/arc/processor/BeanDeployment.java

```diff
@@ -720,20 +720,21 @@ static ScopeInfo getValidScope(Collection<ScopeInfo> stereotypeScopes, Annotatio
             Set<MethodInfo> methods = new HashSet<>();
             while (aClass != null) {
                 if (!scannedClasses.add(aClass)) {
-                    continue;
```

Comment:
continue here would result in an infinite loop,  as aClass will never be updated.

Reply:
Actually this would always return true, as scannedClasses has the wrong scope.

Reply:
> continue here would result in an infinite loop, as aClass will never be updated.

You can't have a class multiple times in the hierarchy, or?

> Actually this would always return true, as scannedClasses has the wrong scope.

I don't think so. We need to use a new set for each bean class.

@michalszynkiewicz could you review this PR? Git blame tells me that you're the author of the original code...

Reply:
I have just removed the scannedClasses set, you can't never have a double up in the hierarchy when you are just looking at superclasses. If there had been a double up with the existing code you would get an infinite loop.

Reply:
While you're at it, could you also remove it here: https://github.com/quarkusio/quarkus/blob/master/independent-projects/arc/processor/src/main/java/io/quarkus/arc/processor/InterceptorInfo.java#L61

Thanks :-)

Reply:
Nevermind, I'll update the interceptor info in a different PR...